ZAPD/ZCollision.cpp

```diff
@@ -291,46 +302,64 @@ CameraDataList::CameraDataList(ZFile* parent, const std::string& prefix,
 	assert(numElements < 10000);
 
 	offset_t cameraPosDataSeg = rawDataIndex;
+	uint32_t numDataTotal;
+	uint32_t cameraPosDataOffset2;
+	uint32_t cameraPosDataSegEnd = rawDataIndex;
+	bool isSharpOcarina = false;
+
 	for (size_t i = 0; i < numElements; i++)
 	{
-		CameraDataEntry* entry = new CameraDataEntry();
+		CameraDataEntry entry;
 
-		entry->cameraSType =
+		entry.cameraSType =
 			BitConverter::ToInt16BE(rawData, rawDataIndex + (entries.size() * 8) + 0);
-		entry->numData = BitConverter::ToInt16BE(rawData, rawDataIndex + (entries.size() * 8) + 2);
-		entry->cameraPosDataSeg =
+		entry.numData = BitConverter::ToInt16BE(rawData, rawDataIndex + (entries.size() * 8) + 2);
+		entry.cameraPosDataSeg =
 			BitConverter::ToInt32BE(rawData, rawDataIndex + (entries.size() * 8) + 4);
 
-		if (entry->cameraPosDataSeg != 0 && GETSEGNUM(entry->cameraPosDataSeg) != SEGMENT_SCENE)
+		if (entry.cameraPosDataSeg != 0 && GETSEGNUM(entry.cameraPosDataSeg) != SEGMENT_SCENE)
 		{
 			cameraPosDataSeg = rawDataIndex + (entries.size() * 8);
 			break;
 		}
 
-		if (entry->cameraPosDataSeg != 0 && cameraPosDataSeg > (entry->cameraPosDataSeg & 0xFFFFFF))
-			cameraPosDataSeg = (entry->cameraPosDataSeg & 0xFFFFFF);
+		if (rawDataIndex > GETSEGOFFSET(entry.cameraPosDataSeg))
+		{
+			if (entry.cameraPosDataSeg != 0 &&
+			    cameraPosDataSeg > GETSEGOFFSET(entry.cameraPosDataSeg))
+				cameraPosDataSeg = GETSEGOFFSET(entry.cameraPosDataSeg);
+		}
+		else  // Sharp Ocarina
+		{
+			isSharpOcarina = true;
+			cameraPosDataOffset2 = rawDataIndex + (numElements * 0x8);
+			cameraPosDataSeg = cameraPosDataOffset2;
```

Comment:
It looks like for Sharp Ocarina, `cameraPosDataSeg` does not get modified after this line? So can `cameraPosDataOffset2` be removed?